### PR TITLE
memory/secure_dump: add test for s390x SE decryption

### PR DESCRIPTION
**Pull request by smitterl**
Add test case that
1. Dumps the SE memory
2. Decrypts the memory
3. Does a minimal check on the decrypted content.

### CODE REVIEW

**Comment by smitterl** [comment]
@fbq815 @liang-cong-red-hat Please help review, thanks!

**Comment by smitterl** [comment]
```
JOB ID     : 0340b49ce3fda53cb68dd9a7a54c846e2c2e5172
JOB LOG    : /var/lib/avocado/job-results/job-2023-12-18T05.24-0340b49/job.log
 (1/1) type_specific.io-github-autotest-libvirt.secure_dump.decrypt: PASS (29.94 s)
Not opening readers, lock_server_running not locked
Not opening readers, lock_server_running not locked
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-12-18T05.24-0340b49/results.html
JOB TIME   : 30.26 s
```

**Comment by fbq815** [comment]
LGTM

File: libvirt/tests/src/memory/secure_dump.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    Decrypt encrypted memory dump.
+    Assumes the guest is running in with
+    SE and has been encrypted with the tested
+    communication key.
+    """
+
+    vm_name = params.get('main_vm')
+    vm = env.get_vm(vm_name)
+    comm_key = params.get("comm_key")
+    dump_file = "/root/DUMP"
```
Comment (chunfuwen) [inline libvirt/tests/src/memory/secure_dump.py:69]: in some case, root folder is not accessible to some users due to permission constrains, so could it be /tmp/DUMP

**Comment by smitterl** [comment]
I had to do some fix for the content check and got drawn away. Please, @chunfuwen could you review again? I also got a LGTM from @fbq815 (s. comment above) so I think you're Approval should be sufficient?

```
JOB ID     : 12291f5b16ba588d79de40853df1a2e9e9fa1f6f
JOB LOG    : /var/lib/avocado/job-results/job-2024-04-01T07.36-12291f5/job.log
 (1/1) type_specific.io-github-autotest-libvirt.secure_dump.decrypt: PASS (19.91 s)
Not opening readers, lock_server_running not locked
Not opening readers, lock_server_running not locked
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2024-04-01T07.36-12291f5/results.html
JOB TIME   : 20.27 s
```

**Review by chunfuwen — APPROVED** [review]
lgtm

**Pull request merged**